test/e2e/e2e_test.go

```diff
@@ -0,0 +1,562 @@
+// Copyright (c) 2020-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+//+build e2e
+
+package e2e
+
+/*
+   The following line makes this package work with LSP in Emacs
+
+	 (setq lsp-go-build-flags ["-tags=e2e"])
+
+   To make this file work properly with LSP in VSCode, set the following in settings.json:
+	 "gopls.env": {
+				"GOFLAGS": "-tags=e2e"
+		},
+*/
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/mattermost/awat/model"
+	cloud "github.com/mattermost/mattermost-cloud/model"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	slackArchive = "../dummy-slack-workspace-archive.zip"
+	slackTeam    = "slack-import"
+)
+
+type environment struct {
+	awatURL     string
+	bucket      string
+	key         string
+	testDomain  string
+	provisioner *cloud.Client
+}
+
+func TestSlackTranslationAndImport(t *testing.T) {
+	env := setupEnv(t)
+	installation := getInstallation(t, env)
+	client := model.NewClient(env.awatURL)
+	ts := startTranslationAndWaitForItToSucceed(t, client, installation, env)
+	waitForImportToSucceed(t, ts, client, env.provisioner, installation)
+
+	t.Log("validate the import data was imported properly")
+
+	clusterInstallations, err := env.provisioner.GetClusterInstallations(
+		&cloud.GetClusterInstallationsRequest{
+			Paging:         cloud.AllPagesNotDeleted(),
+			InstallationID: installation.ID,
+		})
+	require.NoError(t, err)
+	require.True(t, len(clusterInstallations) > 0)
+	ci := clusterInstallations[0]
+
+	checkTeams(t, env.provisioner, ci)
+	checkChannels(t, env.provisioner, ci)
+	checkUsers(t, env.provisioner, ci)
+	checkPosts(t, env.provisioner, ci)
+}
+
+func setupEnv(t *testing.T) *environment {
+	t.Log("validate the environment and gather variables")
+
+	env, err := validatedEnvironment()
+	require.NoError(t, err)
+
+	err = ensureArtifactInBucket(env.bucket)
```

Comment:
It seems like we call this function also in `validatedEnvironment`.
Btw does it make sense to check if the object exists in the bucket if we delete it anyway after the test? Maybe we can just call `PutObject` every time and overwrite the old one if exists?

Reply:
Ah thanks for noticing the double-call, and yeah I think that does make sense, not sure I realized PutObject would overwrite (and testing AWS APIs can be a little bit laborious so please forgive me for not testing that assumption)